mm/2s2h/Rando/ActorBehavior/Souls.cpp

```diff
@@ -10,6 +10,16 @@ extern "C" {
 void func_80B0CF24(BossHakugin*, PlayState*);
 }
 
+bool shouldMajoraRegister() {
+    bool registerStatus = false;
+    if (IS_RANDO && RANDO_SAVE_OPTIONS[RO_SHUFFLE_BOSS_SOULS] == RO_GENERIC_YES) {
+        registerStatus = true;
+    } else if (IS_RANDO && RANDO_SAVE_OPTIONS[RO_SHUFFLE_TRIFORCE_PIECES] == RO_GENERIC_YES) {
+        registerStatus = true;
+    }
+    return registerStatus;
+}
```

Comment:
```suggestion
bool shouldMajoraRegister() {
    bool registerStatus = false;
    if (IS_RANDO) {
        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_BOSS_SOULS] == RO_GENERIC_YES ||
            RANDO_SAVE_OPTIONS[RO_SHUFFLE_TRIFORCE_PIECES] == RO_GENERIC_YES) {
            registerStatus = true;
    }
    return registerStatus;
}
```

Non-blocking question: is this change so that players can't just beat Majora the old fashioned way in Triforce Hunt?

Reply:
Yea, thats the intention. Also a mirror of how SoH handles Ganons Boss Key. I'm not married to this similarity but it seemed to make the most sense since we arent shuffling the pots in Majora's Lair there would be nothing in that room of value.

Reply:
I think you marked this as resolved before applying the suggestion, or do you have a reason for preferring your syntax?
```suggestion
bool shouldMajoraRegister() {
    bool registerStatus = false;
    if (IS_RANDO) {
        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_BOSS_SOULS] == RO_GENERIC_YES ||
            RANDO_SAVE_OPTIONS[RO_SHUFFLE_TRIFORCE_PIECES] == RO_GENERIC_YES) {
            registerStatus = true;
        }
    }
    return registerStatus;
}
```